### PR TITLE
Reduce trait impl duplication and rename DoubleBuff to DoubleBuf

### PR DESCRIPTION
**Pull request by groscoe2**
Partially addresses trait impl duplication mentioned in comment in #113. The duplicate_item macro from the duplicate crate should be sufficient to address the rest of the duplication. I can add that to this PR if desired!

### CODE REVIEW

**Comment by groscoe2** [comment]
@ejmahler I believe this is good to go for review. Now that there are only two duplicate impls for each of AvxArray, SseArray, and NeonArray, I think adding `duplicate` as a dependency is unnecessary. However, it is quite straightforward to eliminate this duplication using that crate if you believe it makes sense to do so.

File: src/avx/avx_mixed_radix.rs [hunk 1]
```diff
@@ -229,17 +229,13 @@ macro_rules! mixedradix_column_butterflies {
...
+                                .load_partial1_complex(partial_remainder_base + len_per_row * i);
                         }
                     } else {
                         for i in 0..ROW_COUNT {
-                            columns[i] = AvxArray::<A>::load_partial2_complex(
-                                &buffer,
-                                partial_remainder_base + len_per_row * i,
-                            );
+                            columns[i] = buffer
+                                .load_partial2_complex(partial_remainder_base + len_per_row * i);
```
Comment (ejmahler) [inline src/avx/avx_mixed_radix.rs:238]: Nice! I always hated having to manually specify the trait here, this is way nicer

File: src/avx/avx_vector.rs [hunk 2]
```diff
@@ -2207,38 +2141,38 @@ impl<T: AvxNum> AvxArrayMut<T> for &mut [Complex<T>] {
...
+impl<'a, T: AvxNum> AvxArrayMut<T> for DoubleBuf<'a, T>
+where
+    Self: AvxArray<T>,
+    &'a mut [Complex<T>]: AvxArrayMut<T>,
+{
     #[inline(always)]
     unsafe fn store_complex(&mut self, data: T::VectorType, index: usize) {
-        debug_assert!(self.output.len() >= index + T::VectorType::COMPLEX_PER_VECTOR);
-        T::VectorType::store_complex(self.output.as_mut_ptr().add(index), data);
+        self.output.store_complex(data, index);
```
Comment (ejmahler) [inline src/avx/avx_vector.rs:2151]: It hadn't occurred to me to just forward the call like this, that's great. This reduces the duplication of the actual business logic too, making the copy/paste issue even less urgent.

File: src/neon/neon_prime_butterflies.rs
Comment (ejmahler) [inline src/neon/neon_prime_butterflies.rs:1]: Did you make any intentional changes in this file? It looks like rustfmt made a whole bunch of changes, to the point where it's hard for me to dig through to find non-formatting changes.

**Comment by ejmahler** [comment]
Looking great. I'm curious what happened with the prime butterfly files, it's like rustfmt only now decided to do anything about them.

File: src/avx/avx_mixed_radix.rs (hunk 1 above)
Comment (groscoe2) [reply src/avx/avx_mixed_radix.rs:238]: Most of these instances were changed already actually! I just found a few more to update

File: src/neon/neon_prime_butterflies.rs
Comment (HEnquist) [reply src/neon/neon_prime_butterflies.rs:1]: We decided to not auto-format the generated code. It breaks the math expressions up in a way that make them very hard to follow. The files that are supposed to be left alone are listed here: https://github.com/ejmahler/RustFFT/blob/4022acc3bac3df360faeb734048611cb00e55f39/rustfmt.toml#L8

Comment (groscoe2) [reply src/neon/neon_prime_butterflies.rs:1]: @HEnquist Gotcha, thank you for the info! It looks like because I am on stable, the rustfmt ignore option is not used, causing my format-on-save setting to mess things up. I can go in and revert the change for that file, but it looks like sse_prime_butterflies.rs was also messed up by rustfmt at some point.

**Comment by groscoe2** [comment]
@ejmahler I've reverted the auto-formatting changes for both prime butterflies files. It turns out that rust nightly is required for the ignore config in rustfmt.toml to work, and I've been using stable for all my development. It would be good to update the contribution section on the readme to reflect this (along with the MSRV increase)!
